commander/src/utils/core/commons.ts

```diff
@@ -158,3 +170,84 @@ export const validateVersion = async (
 		throw new Error(error.message);
 	}
 };
+
+export const getSemver = (str: string): string => {
+	const exp = new RegExp(
+		/(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-(?:[1-9]\d*|[\da-z-]*[a-z-][\da-z-]*)(?:\.(?:[1-9]\d*|[\da-z-]*[a-z-][\da-z-]*))*)?\.?(?:0|[1-9]\d*)?/,
+	);
+	const result = exp.exec(str) as ReadonlyArray<string>;
+
+	return result[0];
+};
+
+export const dateDiff = (date1: Date, date2: Date): number => {
+	const MINUTES_OR_SECONDS = 60;
+	const HOURS = 24;
+	const INT_RANGE = 1000;
+
+	return (
+		(((new Date(date1) as unknown) as number) -
```

Comment:
this conversion is wrong as it's subtracting the Date instances

Reply:
it works https://runkit.com/manugowda/5ccbed7828b3a8001ada0e0c

Reply:
it works then no need to cast the type, but if it's required to cast the type, let's just do it as typescript suggests